### ISSUE TITLE
feature request: `NotNil` and `Nil` recommendations

### ISSUE THREAD

**Issue by dhrubabasu** [Antonboom/testifylint#10]
When `require.NotEqual(nil, a)` is detected, it should recommend `require.NotNil(a)`.

When `require.Equal(nil, a)` is detected, it should recommend `require.Nil(a)`.

**Comment by Antonboom** [comment]
@dhrubabasu, implemented, thanks!

Example for Grafana:
```
$ testifylint --enable error-nil,nil-compare ./... 2>&1 | grep nil-compare
grafana/pkg/components/simplejson/simplejson_test.go:29:2: nil-compare: use assert.NotNil
grafana/pkg/components/simplejson/simplejson_test.go:39:2: nil-compare: use assert.NotNil
grafana/pkg/components/simplejson/simplejson_test.go:49:2: nil-compare: use assert.NotNil
grafana/pkg/components/simplejson/simplejson_test.go:159:2: nil-compare: use assert.NotNil
grafana/pkg/services/publicdashboards/api/api_test.go:278:5: nil-compare: use assert.Nil
```

```go
        js, err := NewJson([]byte(`{"test": {}}`)
	assert.NotEqual(t, nil, js)
```
```go
	aws := js.Get("test").Get("arraywithsubs")
	assert.NotEqual(t, nil, aws)
```
```go
	arr := js.Get("test").Get("array")
	assert.NotEqual(t, nil, arr)
```
```go
	assert.NotEqual(t, nil, val.Params.data)
```
```go
	var jsonResp any
	err := json.Unmarshal(response.Body.Bytes(), &jsonResp)
	require.NoError(t, err)
	assert.Equal(t, jsonResp, nil)
```
